packages/mobile/src/qrcode/QRScanner.tsx

```diff
@@ -1,176 +1,95 @@
-import { RootState } from 'src/redux/reducers'
 import { handleBarcodeDetected, QrCode } from 'src/send/actions'
-import { TransactionDataInput } from 'src/send/SendAmount'
 import Logger from 'src/utils/Logger'
 
-interface StateProps {
-  scanIsForSecureSend?: true
-  transactionData?: TransactionDataInput
-}
-
-interface DispatchProps {
-  handleBarcodeDetected: typeof handleBarcodeDetected
-}
+type Props = StackScreenProps<QRTabParamList, Screens.QRScanner>
 
-type OwnProps = StackScreenProps<StackParamList, Screens.QRScanner> & { isFocused: boolean }
-type Props = DispatchProps & WithTranslation & StateProps & OwnProps
+const SeeThroughOverlay = () => {
+  const { width, height } = variables
 
-const mapStateToProps = (state: RootState, ownProps: OwnProps): StateProps => {
-  const { route } = ownProps
-  return {
-    scanIsForSecureSend: route.params?.scanIsForSecureSend,
-    transactionData: route.params?.transactionData,
-  }
-}
+  const margin = 40
+  const centerBoxSize = width - margin * 2
+  const centerBoxBorderRadius = 8
 
-const mapDispatchToProps = {
-  handleBarcodeDetected,
-}
-
-const QRScannerWrapper = (props: Props) => {
-  const isFocused = useIsFocused()
-  return <QRScanner {...props} isFocused={isFocused} />
+  // TODO(jeanregisser): Investigate why the mask is pixelated on iOS.
+  // It's visible on the rounded corners but since they are small, I'm ignoring it for now.
+  return (
+    <Svg height={height} width={width} viewBox={`0 0 ${width} ${height}`}>
+      <Defs>
+        <Mask id="mask" x="0" y="0" height="100%" width="100%">
+          <Rect height="100%" width="100%" fill="#fff" />
+          <Rect
+            x={margin}
+            y={(height - centerBoxSize) / 2}
+            rx={centerBoxBorderRadius}
+            ry={centerBoxBorderRadius}
+            width={centerBoxSize}
+            height={centerBoxSize}
+            fill="#000"
+          />
+        </Mask>
+      </Defs>
+      <Rect height="100%" width="100%" fill="rgba(0,0,0,0.5)" mask="url(#mask)" />
+    </Svg>
+  )
 }
 
-class QRScanner extends React.Component<Props> {
-  static navigationOptions = () => ({
-    ...headerWithBackButton,
-    headerTitle: i18n.t('sendFlow7:scanCode'),
-  })
+export default function QRScanner({ route }: Props) {
+  const dispatch = useDispatch()
+  const { t } = useTranslation(Namespaces.sendFlow7)
+  const inset = useSafeArea()
 
-  camera: RNCamera | null = null
+  const { scanIsForSecureSend, transactionData } = route.params || {}
 
-  // This method would be called multiple times with the same
-  // QR code, so we need to catch only the first one
-  onBarCodeDetected = memoize(
-    (qrCode: QrCode) => {
-      Logger.debug('QRScanner', 'Bar code detected')
-      const { scanIsForSecureSend, transactionData } = this.props
-      this.props.handleBarcodeDetected(qrCode, scanIsForSecureSend, transactionData)
-    },
-    (qrCode) => qrCode.data
+  const onBarCodeDetected = useCallback(
+    memoize(
```

Comment:
Does this work? Or do we create a new memoized function on each render?

Reply:
yeah im not sure i get why its memoized.

Reply:
Yes not very obvious on the first look indeed.

It does create a new memoized version on each render, but only the first memoized version is returned, unless `[scanIsForSecureSend, transactionData]` change.

We could think calling `memoize` on each render seems heavy, but it's not very different from creating a new function in a standard `useCallback` call.

See https://github.com/lodash/lodash/blob/master/memoize.js#L43-L60